templates/vdbench_save.yml

```
@@ -0,0 +1,181 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# https://aka.ms/yaml
+
+parameters:
+- name: vdbench_config
+  type: string
+
+steps:
+- script: |
+    export PATH=$PATH:${BUILD_SOURCESDIRECTORY}/test/utils
+
+    az login --service-principal -u $ARM_CLIENT_ID -p $ARM_CLIENT_SECRET --tenant $ARM_TENANT_ID &&
+    az account set -s $ARM_SUBSCRIPTION_ID
+    check_rc.sh $? "Login to Azure and set default subscription" || exit 0
+
+    # Wait for the vdbench.py script to complete.
+    cmdokmsg="CMD_OK_$(date -u '+%m%dx%H%M%S')"
+    cmdlog=/tmp/vm_runcmd.log
+    polling_secs=60
+    while [ 1 ]; do
+      az vmss run-command invoke                      \
+        --resource-group ${RESOURCE_GROUP_NAME}-vmss  \
+        --name           "vmss"                       \
+        --instance-id    0                            \
+        --command-id     RunShellScript               \
+        --scripts        "runuser -l azureuser -c 'tmux ls' || echo ${cmdokmsg}" \
+        --query 'value[0].message' --output tsv |& tee ${cmdlog}
+      check_rc.sh ${PIPESTATUS[0]} "Checking for any tmux session " || exit
+      grep -q ${cmdokmsg} ${cmdlog}
+      if [ $? -ne 0 ]; then
+        echo "INFO: tmux session still active. Re-check in ${polling_secs} seconds."
+        sleep ${polling_secs}
+        continue
+      else
+        echo "INFO: tmux session no longer active"
+        break
+      fi
+    done
+
+    cmdokmsg="CMD_OK_$(date -u '+%m%dx%H%M%S')"
+    cmdlog=/tmp/vm_runcmd_tail.log
+    az vmss run-command invoke                     \
+      --resource-group ${RESOURCE_GROUP_NAME}-vmss \
+      --name           "vmss"                      \
+      --instance-id    0                           \
+      --command-id     RunShellScript              \
+      --scripts        "runuser -l azureuser -c 'tail -25 /home/azureuser/vdb.log' && echo ${cmdokmsg}" \
+      --query 'value[0].message' --output tsv |& tee ${cmdlog}
+    grep -q ${cmdokmsg} ${cmdlog} || check_rc.sh 1 "Tail vdbench.log"
+
+    grep -q 'Vdbench execution completed successfully' ${cmdlog}
+    grep_rc=$?
+    check_rc.sh ${grep_rc} "vdbench.py success check"
```

Not a must-change, but you could grep the whole file via run-command. If found, then you'll get command OK. Something like:
```suggestion
      --scripts        "runuser -l azureuser -c 'grep 'Vdbench execution completed successfully' /home/azureuser/vdb.log' && echo ${cmdokmsg}" \
      --query 'value[0].message' --output tsv |& tee ${cmdlog}
    grep -q ${cmdokmsg} ${cmdlog} || check_rc.sh 1 "Tail vdbench.log"
```

---

Hmm... the three layers of quoting might make this more trouble than it's worth. Your call.

Maybe something regex-y like `grep Vdbench.execution.completed.successfully /home/azureuser/vdb.log` ... where the `.` will match anything, including spaces.

---

I like how it shows the output in the pipeline, if you are ok with leaving as is, I think I would prefer this output.
Pipeline Runs:
- https://dev.azure.com/averevfxt/vfxt-github/_build/results?buildId=9104&view=logs&j=95dcbc90-489f-5462-2295-bdc9b9833630&t=7aa6da0e-b524-5ff4-02cc-53a9e81bab8b&l=109
- https://dev.azure.com/averevfxt/vfxt-github/_build/results?buildId=9105&view=logs&j=95dcbc90-489f-5462-2295-bdc9b9833630&t=4eb5a689-45d6-566d-45aa-8f36c2fafffc&l=109